packages/core/src/entityStatement/fetchEntityStatementChain.ts

```diff
@@ -0,0 +1,65 @@
+import type { EntityConfigurationClaims } from '../entityConfiguration'
+import { ErrorCode } from '../error/ErrorCode'
+import { OpenIdFederationError } from '../error/OpenIdFederationError'
+import type { VerifyCallback } from '../utils'
+import type { EntityStatementClaims } from './entityStatementClaims'
+import { fetchEntityStatement } from './fetchEntityStatement'
+
+export type FetchEntityStatementChainOptions = {
+  entityConfigurations: Array<EntityConfigurationClaims>
+  verifyJwtCallback: VerifyCallback
+}
+
+export const fetchEntityStatementChain = async ({
+  verifyJwtCallback,
+  entityConfigurations,
+}: FetchEntityStatementChainOptions) => {
+  // The trust anchors (i.e. the last item of the list) entity configuration will be used instead of a statement issued by a superior
+  const statements = [entityConfigurations[entityConfigurations.length - 1]]
+
+  // Reverse the configurations as we have to fetch the other way around
+  // A copy is done here as the reverse method  mutates in place
+  const reversedConfigurations = [...entityConfigurations].reverse()
+
+  const promises: Array<Promise<EntityStatementClaims>> = []
+
+  for (let i = 0; i < reversedConfigurations.length; i++) {
+    const configuration = reversedConfigurations[i]
+    const subjectConfiguration = reversedConfigurations[i + 1]
+
+    if (!subjectConfiguration) continue
+
+    const fetchEndpoint = configuration.source_endpoint
+
+    if (!fetchEndpoint) {
+      throw new OpenIdFederationError(
+        ErrorCode.Validation,
+        `No source endpoint found for configuration for: '${configuration.sub}'`
+      )
+    }
+
+    const iss = configuration.iss
+    const sub = subjectConfiguration.sub
+
+    promises.push(
+      fetchEntityStatement({
+        verifyJwtCallback,
+        endpoint: fetchEndpoint,
+        iss,
+        sub,
+        issEntityConfiguration: configuration,
+      })
+    )
+  }
+
+  const results = await Promise.allSettled(promises)
+  for (const res of results) {
+    if (res.status === 'fulfilled') {
+      statements.push(res.value)
+    } else {
+      throw new OpenIdFederationError(ErrorCode.Validation, res.reason)
+    }
+  }
```

Comment:
If it fails when one fails, we can just use Promise.all right?

Would be good to pass an abort signal to the fetch methods so you can cancel all promises instantly when one fails.

Reply:
Created a ticket for the abort signal. Optimalization for later I'd say.